### ISSUE TITLE
Discussion: Hosting providers for OME

### ISSUE THREAD

**Issue by llspalex** [AirenSoft/OvenMediaEngine#481]
I saw a lot of mention of DigitalOcean in #468 and wanted to mention a few possible alternatives for OME users. 

- Linode. Pricing is very similar (and for dedicated vCPU instances can be cheaper), it has very generous data transfer allowances and - importantly for OME users - publishes their 'network out' speeds. Though I think their 'network out' speeds are the maximum, not guaranteed speeds, because I was getting 700mbps out during some tests on an instance with advertisted network out of 4,000 mbps.

- Vultr. Similar to Linode / DO with their generous data transfer allowances. I mention them specifically because when I ran speedtests (using [Speedtest-CLI](https://www.speedtest.net/apps/cli)) I was getting in excess of 4,500mbps in and 9,000 mbps out on an instance hosted in Australia. But I haven't done any thorough network speed tests with them (the speeds from my one test may have been a one-off).

- Hetzner. Locations only in Germany and Finland, but they are _very_ affordable with seriously big data transfer allowances. Not sure what their network speed is like. 

Keep in mind that if you are looking at the built-in GUI CPU usage graphs on Linode or Vultr - they report CPU usage per core (ie. 100% usage on a 2 vCPU instance = 50% overall usage), whereas DO's built-in graphs report overall CPU usage (ie. 100% usage on a 2 vCPU instance = 100% overall usage). 

Does anyone know if there are hosting providers that have guaranteed network out speeds? 

@fcqpl know you were looking into low-cost hosting options, so you may be interested in this.

**Comment by mpisat** [comment]
I got a dedi server from ikoula for 44.09 EUR a month after 30% lifetime discount, I7-9700K (8 real cores, Quicksync enabled!), 960GB NVME, 64GB Ram.
Network speed is pretty good in the EU.

**Comment by MoZyo** [comment]
[OVH](ovhcloud.com) :)

**Comment by basisbit** [comment]
OVH: Beware traffic costs: 100$ per TB for their Singapore servers.

Vultr: Beware traffic costs: 10$ per TB for North America and Western Europe; 25$ per TB for Singapore and Japan; 50$ per TB for Australia and South Korea

Hetzner: 1$ per TB of traffic after using the initial 20TB of free traffic included with each VM per month, or use a dedicated server with "unlimited" included traffic. They occasionally kick very big customers if they don't play fair. Typical reliable (lowest) network speed for Hetzner Cloud VMs: 400Mb/s, for their dedicated servers: 650Mb/s

IONOS: unlimited free traffic, but their VMs have very unreliable network upload speeds (often below 150Mb/s), do not use. Their dedicated servers (hourly billed) have mediocre network upload speed stability up to ~ 500Mb/s

DO: 10$ per TB of traffic for not-included traffic, or 5$ per TB for included traffic (order bunch of unused shared resource VMs when you are close to the end of the month to turn 10$/TB traffic into 5$/TB traffic), and all their servers have very unreliable network upload speeds, especially in Asia and USA. Sometimes 800Mb/s, sometimes 150Mb/s. Upload speed is independent from "droplet" size, thus I'd suggest using many smaller instance sizes instead of a few big ones. 

Linode: 10$ per TB over too much traffic usage, but they kick customers who use a lot of traffic on a lot of servers. Also rather expensive for the amount of CPU resources.

**Comment by naanlizard** [comment]
@basisbit do you have any recommendations for an edge server with moderate bandwidth usage (~30TiB/mo) in the SEA/Asia region? Data seems ridiculously expensive in the region

**Comment by basisbit** [comment]
Not really. Maybe try Oracle Cloud or try to find a dedicated server provider that bills traffic per TB and which buys enough transit from (preferably all) big Tier 1 providers of that region.

**Comment by mpisat** [comment]
> @basisbit do you have any recommendations for an edge server with moderate bandwidth usage (~30TiB/mo) in the SEA/Asia region? Data seems ridiculously expensive in the region

Singapore has the best connectivity to SEA but don't expect cheap BW :)
Maybe you get 3 of those for 30TB?
https://www.ovhcloud.com/en-sg/bare-metal/rise/rise-apac-1/

**Comment by basisbit** [comment]
Sorry, but OVH's Singapore location is a pita to use. I do not recommend it. Also, their pricing per TB is just a rip-off or them saying they don't want customers in that region.

On a side note, I also have to revert my suggestion of trying Oracle - their business processes are just absolutely terrible, especially regarding sign-up process, trying to get quota canges to be able to create some more servers and trying to get any support that doesn't just tell you "Sorry, but we can't help you". It seems Oracle just doesn't want people to actually use their service.
